src/modules/MessageSearch/context/__test__/MessageSearchProvider.spec.tsx

```diff
@@ -0,0 +1,157 @@
+            next: jest.fn().mockResolvedValue([{ messageId: 1 }]),
+          })),
+        },
+        initialized: true,
+      },
+    },
+    config: { logger: console },
+  })),
+}));
+
+jest.mock('../hooks/useSetChannel', () => ({
+  __esModule: true,
+  default: jest.fn().mockImplementation(() => Promise.resolve()),
+}));
+
+jest.mock('../hooks/useGetSearchedMessages', () => ({
+  __esModule: true,
+  default: jest.fn(() => jest.fn()),
+}));
+
+jest.mock('../hooks/useScrollCallback', () => ({
+  __esModule: true,
+  default: jest.fn(() => jest.fn()),
+}));
+
+jest.mock('../hooks/useSearchStringEffect', () => ({
+  __esModule: true,
+  default: jest.fn(() => ''),
+}));
+
+describe('MessageSearchProvider', () => {
+  it('provides the correct initial state', () => {
+    const wrapper = ({ children }) => (
+      <MessageSearchProvider channelUrl="test-channel">
+        {children}
+      </MessageSearchProvider>
+    );
+
+    const { result } = renderHook(() => useMessageSearchContext(), { wrapper });
+
+    expect(result.current.getState()).toEqual(expect.objectContaining({
+      channelUrl: 'test-channel',
+      allMessages: [],
+      loading: false,
+      isInvalid: false,
+      initialized: false,
+      currentChannel: null,
+      currentMessageSearchQuery: null,
+      hasMoreResult: false,
+    }));
+  });
+
+  it('updates state correctly when props change', () => {
+    const wrapper = ({ children }) => (
+      <MessageSearchProvider channelUrl="test-channel">
+        {children}
+      </MessageSearchProvider>
+    );
+
+    const { result } = renderHook(() => useMessageSearchContext(), { wrapper });
+    expect(result.current.getState().channelUrl).toBe('test-channel');
+    result.current.setState({ channelUrl: 'new-channel' });
+    expect(result.current.getState().channelUrl).toBe('new-channel');
+  });
+
+  it('provides correct actions through useMessageSearch hook', () => {
+    const wrapper = ({ children }) => (
+      <MessageSearchProvider channelUrl="test-channel">
+        {children}
+      </MessageSearchProvider>
+    );
+
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    expect(result.current.actions).toHaveProperty('setCurrentChannel');
+    expect(result.current.actions).toHaveProperty('getSearchedMessages');
+    expect(result.current.actions).toHaveProperty('startMessageSearch');
+  });
+
+  it('updates state correctly when actions are called', async () => {
+    const wrapper = ({ children }) => (
+      <MessageSearchProvider channelUrl="test-channel">
+        {children}
+      </MessageSearchProvider>
+    );
+
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    await act(async () => {
+      result.current.actions.setCurrentChannel({ url: 'test-channel' } as any);
+      await waitFor(() => {
+        expect(result.current.state.currentChannel).toEqual({ url: 'test-channel' });
+        expect(result.current.state.initialized).toBe(true);
```

Comment:
It makes more sense to me to make sure the other states such as `loading` and `isInvalid` unchanged. It would help finding some regression before the thing could go into trouble imho.

Reply:
Thanks. So I updated other tests as well https://github.com/sendbird/sendbird-uikit-react/pull/1228/commits/b6849336baa909aa9164660b492eea369d9644be